internal/services/recommendation_poller.go

```diff
@@ -15,112 +16,126 @@ import (
+func commitKafkaMsg(msg *kafka.Message, consumer_object *kafka.Consumer) {
+	_, err := consumer_object.CommitMessage(msg)
+	if err != nil {
+		log.Error("unable to commit msg: ", err)
+	}
+}
+
+func requestAndSaveRecommendation(kafkaMsg types.RecommendationKafkaMsg, recommendationType string) bool {
+	log := logging.GetLogger()
+	experiment_name := kafkaMsg.Metadata.Experiment_name
+	maxEndTimeFromReport := kafkaMsg.Metadata.Max_endtime_report
+	var poll_cycle_complete bool = false
+
+	recommendation, err := kruize.Update_recommendations(experiment_name, maxEndTimeFromReport)
+	if err != nil {
+		end_interval := utils.ConvertDateToISO8601(maxEndTimeFromReport.String())
+		if err.Error() == fmt.Sprintf("Recommendation for timestamp - \" %s \" does not exist", end_interval) {
+			log.Infof("recommendation does not exist for timestamp - \" %s \"", end_interval)
+		}
+		log.Errorf("unable to list recommendation for: %v", err)
+	}
+
+	// TODO: Is_valid_recommendation to be called on every container record v20.1 upgrade on wards
+	if kruize.Is_valid_recommendation(recommendation) {
+		containers := recommendation[0].Kubernetes_objects[0].Containers
+		for _, container := range containers {
+			for _, v := range container.Recommendations.Data {
+				marshalData, err := json.Marshal(v)
+				if err != nil {
+					log.Errorf("unable to list recommendation for: %v", err)
+				}
+
+				// Create RecommendationSet entry into the table.
+				recommendationSet := model.RecommendationSet{
+					WorkloadID:          kafkaMsg.Metadata.Workload_id,
+					ContainerName:       container.Container_name,
+					MonitoringStartTime: v.Duration_based.Short_term.Monitoring_start_time,
+					MonitoringEndTime:   v.Duration_based.Short_term.Monitoring_end_time,
+					Recommendations:     marshalData,
+				}
+				if err := recommendationSet.CreateRecommendationSet(); err != nil {
+					log.Errorf("unable to save a record into recommendation set: %v. Error: %v", recommendationSet, err)
+				} else {
+					log.Infof(" %s - Recommendation saved for experiment - %s and end_interval - %s", recommendationType, experiment_name, recommendationSet.MonitoringEndTime)
+				}
+
+				// Create entry into HistoricalRecommendationSet table.
+				historicalRecommendationSet := model.HistoricalRecommendationSet{
+					OrgId:               kafkaMsg.Metadata.Org_id,
+					WorkloadID:          kafkaMsg.Metadata.Workload_id,
+					ContainerName:       container.Container_name,
+					MonitoringStartTime: v.Duration_based.Short_term.Monitoring_start_time,
+					MonitoringEndTime:   v.Duration_based.Short_term.Monitoring_end_time,
+					Recommendations:     marshalData,
+				}
+				if err := historicalRecommendationSet.CreateHistoricalRecommendationSet(); err != nil {
+					recommendationJSON, _ := json.Marshal(recommendation)
+					log.Errorf("unable to get or add record to historical recommendation set table: %s. Error: %v", string(recommendationJSON), err)
```

Comment:
Same here as well? I think if we are unable to save the recommendations then also we should not commit the message?

Reply:
Ditto

Reply:
Ditto